File: src/main/java/com/mpe85/grampa/input/LineCounter.java
package com.mpe85.grampa.input;

/**
 * A counter for lines in texts.
 * 
 * @author mpe85
 *
 */
public interface LineCounter {
	
	/**
	 * Gets the line count.
	 * 
	 * @return the line count
	 */
	int getLineCount();
	
	/**
	 * Gets the position of a character at a given index.
	 * 
	 * @param index
	 *            a valid index
	 * @return the input position
	 */
	InputPosition getPosition(int index);
	
}

File: src/main/kotlin/com/mpe85/grampa/input/LineCounter.kt
package com.mpe85.grampa.input

/**
 * A counter for lines in texts.
 *
 * @author mpe85
 * @property[lineCount] The line count
 */
interface LineCounter {

    val lineCount: Int

    /**
     * Get the position of a character at a given index.
     *
     * @param[index] A valid index
     * @return The input position at the given index
     */
    fun getPosition(index: Int): InputPosition

}
